Sources/EduIDExpansion/Controllers/Coordinators/MainCoordinator.swift

```diff
@@ -1,17 +1,70 @@
 import UIKit
 
-class MainCoordinator: CoordinatorType {
+class MainCoordinator: CoordinatorType {    
+    
+    weak var parent: CoordinatorType?
     
     var children: [CoordinatorType] = []
-    var navigationController: UINavigationController
+    weak var homeNavigationController: UINavigationController!
     
-    init(navigationController: UINavigationController) {
-        self.navigationController = navigationController
+    init(homeNavigationController: UINavigationController) {
+        self.homeNavigationController = homeNavigationController
+        
     }
     
     func start() {
-        let onboardingCoordinator = OnboardingCoordinator(navigationController: self.navigationController)
+        let onboardingNavigationController = UINavigationController()
+        onboardingNavigationController.modalTransitionStyle = .flipHorizontal
+        onboardingNavigationController.isModalInPresentation = true
+        
+        let onboardingCoordinator = OnboardingCoordinator(navigationController: onboardingNavigationController)
+        onboardingCoordinator.parent = self
         children.append(onboardingCoordinator)
+        
+        homeNavigationController.present(onboardingNavigationController, animated: false)
         onboardingCoordinator.start()
     }
+    
+    func showPersonalInfo() {
+        let editPersonalInfoViewcontroller = EditPersonalInfoViewController()
+        editPersonalInfoViewcontroller.screenType = .personalInfoScreen
+        editPersonalInfoViewcontroller.modalTransitionStyle = .flipHorizontal
+        
+        let personalInfoNavigationController = UINavigationController(rootViewController: editPersonalInfoViewcontroller)
+        personalInfoNavigationController.modalTransitionStyle = .flipHorizontal
+        personalInfoNavigationController.isModalInPresentation = true
+        
+        let personalInfoCoordinator = PersonalInfoCoordinator(navigationController: personalInfoNavigationController)
```

Comment:
The PersonalInfoCoordinator should be responsible for creating the EditPersonalInfoViewController and in this case probably also the UINavigationController, since it seems to present it modally. What you'd pass to the PersonalInfoCoordinator is the view controller on which the presentation would happen (or the navigation controller onto which the flow would be pushed).

Reply:
I have done this